integration/spark/src/main/java/org/apache/carbondata/spark/load/CarbonLoaderUtil.java

```diff
@@ -1204,12 +1208,14 @@ private static void assignLeftOverBlocks(Map<String, List<Distributable>> output
 
     if (activeNodes != null) {
       for (String activeNode : activeNodes) {
-        List<Distributable> blockLst = outputMap.get(activeNode);
```

Comment:
Don't use getBlockList here, it costly. Instead in previous output map population logic, always populate active node name only.